data/mods/gen7/random-teams.ts

```diff
@@ -602,121 +701,105 @@ export class RandomGen7Teams extends RandomGen8Teams {
-		case 'Battle Bond': case 'Dazzling': case 'Flare Boost': case 'Hyper Cutter':
-		case 'Ice Body': case 'Innards Out': case 'Moody': case 'Steadfast': case 'Magician':
+		case 'Battle Bond': case 'Dazzling': case 'Flare Boost': case 'Gluttony': case 'Harvest': case 'Hyper Cutter':
+		case 'Ice Body': case 'Innards Out': case 'Liquid Voice': case 'Magician': case 'Moody': case 'Sand Veil':
+		case 'Snow Cloak': case 'Steadfast': case 'Weak Armor':
 			return true;
 		case 'Aerilate': case 'Galvanize': case 'Pixilate': case 'Refrigerate':
-			return !counter.get('Normal');
-		case 'Analytic': case 'Download':
-			return species.nfe;
-		case 'Battle Armor': case 'Sturdy':
-			return (!!counter.get('recoil') && !counter.get('recovery'));
+			return ['doubleedge', 'hypervoice', 'return'].every(m => !moves.has(m));
 		case 'Chlorophyll':
+			// Petal Dance is for Lilligant
 			return (
-				species.baseStats.spe > 100 ||
-				abilities.has('Harvest') ||
+				species.baseStats.spe > 100 || moves.has('petaldance') ||
 				(!moves.has('sunnyday') && !teamDetails.sun)
 			);
 		case 'Competitive':
-			return (!counter.get('Special') || moves.has('sleeptalk') && moves.has('rest'));
+			return !counter.get('Special');
 		case 'Compound Eyes': case 'No Guard':
-			return !counter.get('inaccurate');
-		case 'Contrary': case 'Iron Fist': case 'Skill Link': case 'Strong Jaw':
+			// Shadow Punch bit is for Golurk
+			return (!counter.get('inaccurate') || moves.has('shadowpunch'));
+		case 'Contrary': case 'Skill Link': case 'Strong Jaw':
 			return !counter.get(toID(ability));
 		case 'Defiant': case 'Justified': case 'Moxie':
-			return !counter.get('Physical') || moves.has('dragontail');
-		case 'Flash Fire':
-			return abilities.has('Drought');
-		case 'Gluttony':
-			return !moves.has('bellydrum');
-		case 'Harvest':
-			return abilities.has('Frisk');
+			return !counter.get('Physical');
 		case 'Hustle':
 			return counter.get('Physical') < 2;
 		case 'Hydration': case 'Rain Dish': case 'Swift Swim':
 			return (
 				species.baseStats.spe > 100 || !moves.has('raindance') && !teamDetails.rain ||
 				!moves.has('raindance') && ['Rock Head', 'Water Absorb'].some(abil => abilities.has(abil))
 			);
-		case 'Slush Rush': case 'Snow Cloak':
-			return !teamDetails.hail;
-		case 'Immunity': case 'Snow Warning':
-			return (moves.has('facade') || moves.has('hypervoice'));
 		case 'Intimidate':
-			return (moves.has('bodyslam') || moves.has('rest') || abilities.has('Reckless') && counter.get('recoil') > 1);
+			// Slam part is for Tauros
+			return (moves.has('bodyslam') || species.id === 'staraptor');
+		case 'Iron Fist':
+			// Dynamic Punch bit is for Golurk
+			return (!counter.get(toID(ability)) || moves.has('dynamicpunch'));
 		case 'Lightning Rod':
 			return (
-				species.types.includes('Ground') ||
-				(!!teamDetails.rain || moves.has('raindance')) && abilities.has('Swift Swim')
+				types.has('Ground') || species.id === 'marowakalola' ||
```

Comment:
In this method, there's an odd mixture of hardcoded species IDs and broader conditions that are noted in comments as being specific to one species.

Reply:
We have used hardcoded `species.id` checks when necessary - however, if there was no need or benefit to have those `species.id` checks, we did not include them, and instead just made a note of it. For example, we want to reject No Guard on Golurk sets with Shadow Punch (but not its other set with Dynamic Punch), so we rejected No Guard with Shadow Punch. There is no need to have a `species.id` check since Golurk is the only Pokemon that can have both.

Reply:
PS contributing guidelines [generally prefer](https://github.com/smogon/pokemon-showdown/blob/master/CONTRIBUTING.md#c2-document-in-names-if-possible) that we document things within the code rather than through comments, if possible.

I think what you describe is acceptable as long as the move/ability/etc checks actually matter within the species and aren't just a stand-in for checking the species (and wouldn't be something you care about on other species).